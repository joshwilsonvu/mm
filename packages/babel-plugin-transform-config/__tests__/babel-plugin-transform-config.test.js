'use strict';

const pluginTester = require('babel-plugin-tester').default;
const plugin = require('../lib/babel-plugin-transform-config');

pluginTester({
  plugin: plugin,
  tests: {
    "adds import": {
      code: `
        const config = {
          port: 8080,
          modules: [
            {
              module: "helloworld",
              position: "top_left",
              config: {
                text: "Hello world"
              }
            },
            {
              module: "MMM-clock",
              position: "top_right"
            }
          ]
        };
        export default config;
      `,
      output: `
        const config = {
          port: 8080,
          modules: [
            {
              module: "helloworld",
              _import: () => import("../modules/default/helloworld/helloworld"),
              position: "top_left",
              config: {
                text: "Hello world"
              }
            },
            {
              module: "MMM-clock",
              _import: () => import("../modules/MMM-clock/MMM-clock"),
              position: "top_right"
            }
          ]
        };
        export default config;
      `
    },
  },
  babelOptions: {
    filename: require.resolve("./config/dummy"),
    babelrc: false,
    configFile: false,
    root: __dirname,
  },
});
