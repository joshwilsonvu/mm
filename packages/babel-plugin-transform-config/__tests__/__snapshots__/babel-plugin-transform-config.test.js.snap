// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-transform-config adds _component, _path to modules: adds _component, _path to modules 1`] = `

export default {
  port: 8080,
  modules: [
    {
      module: "a-module",
      position: "top_right"
    }
  ]
};

      ↓ ↓ ↓ ↓ ↓ ↓

export default {
  port: 8080,
  modules: [
    {
      module: "a-module",
      _component: require("react").lazy(() =>
        import(
          "<PROJECT_ROOT>/packages/babel-plugin-transform-config/__tests__/modules/a-module/a-module.ts"
        )
      ),
      _path:
        "<PROJECT_ROOT>/packages/babel-plugin-transform-config/__tests__/modules/a-module/a-module.ts",
      position: "top_right",
    },
  ],
};


`;

exports[`babel-plugin-transform-config resolves default modules: resolves default modules 1`] = `

export default {
  modules: [
    {
      module: "a-default-module"
    }
  ]
};

      ↓ ↓ ↓ ↓ ↓ ↓

export default {
  modules: [
    {
      module: "a-default-module",
      _component: require("react").lazy(() =>
        import(
          "<PROJECT_ROOT>/packages/babel-plugin-transform-config/__tests__/modules/default/a-default-module/a-default-module.js"
        )
      ),
      _path:
        "<PROJECT_ROOT>/packages/babel-plugin-transform-config/__tests__/modules/default/a-default-module/a-default-module.js",
    },
  ],
};


`;
